src/content/docs/release-notes/agent-release-notes/go-release-notes/go-agent-3-12-0.mdx

```diff
@@ -3,6 +3,9 @@ subject: Go agent
 releaseDate: '2021-05-03'
 version: 3.12.0
 downloadLink: 'https://github.com/newrelic/go-agent/tree/v3.12.0'
+features: []
```

Comment:
FWIW, you don't need to add the field if it's empty. unless the service consuming the JSON expects an empty array. if the field isn't present we resolve it to `null` in the JSON that's generated for these.

Reply:
Our internal spec sheet for this change states that we must add the fields with an empty array. I'll reach out to see if any services we are using expect an array or not. Thanks for pointing this out!

Reply:
@roadlittledawn - Thanks for chiming in! Are there performance or other implications caused by adding fields that hold empty values? (like the build is slower because extra unnecessary fields are made?)

Reply:
@kaylareopelle not 100% sure, but my take is there's likely little to no performance impact of adding fields with empty values

Reply:
Ok, given the lack of negligible performance impact, I think we'll leave the current spec as-is since some teams have already merged their PRs.